src/pages/task/TestsTableCore.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useEffect } from "react";
+import { ColumnProps, TableProps } from "antd/es/table";
+import { InfinityTable } from "antd-table-infinity";
+import { msToDuration } from "utils/string";
+import { Tag, Spin } from "antd";
+import { TESTS_QUERY } from "utils/gql/queries";
+import { useParams, useLocation, useHistory } from "react-router-dom";
+import { useQuery } from "@apollo/react-hooks";
+import { usePrevious } from "utils/hooks";
+import {
+  Categories,
+  RequiredQueryParams,
+  Sort,
+  ValidInitialQueryParams,
+  Limit
+} from "pages/task/types";
+import get from "lodash.get";
+import queryString from "query-string";
+
+interface TaskTests {
+  id: string;
+  status: string;
+  testFile: string;
+  duration: number;
+}
+
+interface UpdateQueryArg {
+  taskTests: [TaskTests];
+}
+
+const loadMoreContent = (): JSX.Element => (
+  <div
+    style={{
+      textAlign: "center",
+      paddingTop: 40,
+      paddingBottom: 40,
+      border: "1px solid #e8e8e8"
+    }}
+  >
+    <Spin tip="Loading..." />
+  </div>
+);
+const columns: Array<ColumnProps<TaskTests>> = [
+  {
+    title: "Name",
+    dataIndex: "testFile",
+    key: Categories.TestName,
+    sorter: true
+  },
+  {
+    title: "Status",
+    dataIndex: "status",
+    key: Categories.Status,
+    sorter: true,
+    render: (tag: string): JSX.Element => {
+      const color = tag === "pass" ? "green" : "geekblue";
+      return (
+        <span>
+          <Tag color={color} key={tag}>
+            {tag.toUpperCase()}
+          </Tag>
+        </span>
+      );
+    }
+  },
+  {
+    title: "Time",
+    dataIndex: "duration",
+    key: Categories.Duration,
+    sorter: true,
+    render: (text: number): string => {
+      return msToDuration(text.toFixed());
+    }
+  }
+];
+const rowKey = ({ id }) => id;
+type Props = ValidInitialQueryParams & Limit;
+export const TestsTableCore: React.FC<Props> = ({
+  initialSort,
+  initialCategory,
+  limit
+}) => {
+  const { taskID } = useParams();
+  const { search, pathname } = useLocation();
+  const { replace } = useHistory();
+  const { data, fetchMore, networkStatus } = useQuery(TESTS_QUERY, {
+    variables: {
+      id: taskID,
+      dir: initialSort === Sort.Asc ? "ASC" : "DESC",
+      cat: initialCategory,
+      pageNum: 0,
+      limitNum: limit
+    },
+    notifyOnNetworkStatusChange: true
+  });
+  const parsed = queryString.parse(search);
+  const category = (parsed[RequiredQueryParams.Category] || "")
+    .toString()
+    .toUpperCase();
+  const sort = parsed[RequiredQueryParams.Sort];
+  const prevCategory = usePrevious(category);
+  const prevSort = usePrevious(sort);
+  useEffect(() => {
+    const elements = document.querySelectorAll(
```

Comment:
Instead of doing this, can you style the header cell in the column definitions? This feels pretty janky

Reply:
agreed. this is an unideal way of disabling the sort buttons while a query is loading

Reply:
The column prop provides a way for us to hide the sorter buttons all together while a network request is in progress but doesn't allow a way to disable them. Hiding the buttons all together is confusing for the end user when the network request is taking a long time.
I disabled the sort buttons in the first place was because I was unsure if the data was rendering correctly in the table when there were multiple requests being made at the same time. Part of that issue is duplicate keys being returned from tests endpoint between 2 different pages which I'm debugging so there is a chance we may not need to disable the sorters.

Reply:
I didn't realize this was the motivation behind that, but if that's the case should we have a consistent way to block input while network requests are in progress throughout the app? For example we can show a spinner which also visually and functionally disables the components below it. That would solve the problem of trying to interact while there is a network request, and also prevent a certain class of race conditions that are possible by launching another network request while one is in progress

Reply:
i don't think we need a full-on universal solution for this occurrence just yet. solving it for this use case first is sufficient imo. As more examples of where a consistent pattern occur we can then go ahead and design an implementation. We will also better understand _how_ to design something like this well when we understand more use cases.

Reply:
i'm more so surprised that the antd infinite scroll component does not have a `disable` prop for the header actions. the components in the ant lib itself usually have this functionality. Speaking of blocking input while network requests are in progress, this is how that should be accomplished.

Reply:
What if the event handler you pass to `onFetch` is bound to the context of `TestsTableCore`. Then in the event handler you can do something like this: 
```
const {data, loading} = useQuery(QUERY)

const onFetch = () => {
    if (!loading) {
        ...fetch stuff...
    }
});
``

Reply:
@tgrander That solution works to prevent network requests but the sorting button will still toggle and misrepresent the data being shown.

Reply:
I'm going to look into this further